protocol/communities/community.go

```diff
@@ -1371,6 +1374,40 @@ func (o *Community) tokenPermissions() map[string]*CommunityTokenPermission {
 	return result
 }
 
+func (o *Community) PendingAndBannedMembers() map[string]CommunityMemberState {
+	result := make(map[string]CommunityMemberState)
+
+	// Non-privileged members should not see pending and banned members
+	if o.config.EventsData == nil || !o.IsPrivilegedMember(o.MemberIdentity()) {
+		return result
+	}
+
+	for _, bannedMemberID := range o.config.CommunityDescription.BanList {
+		result[bannedMemberID] = CommunityMemberBanned
+	}
+
+	processedEvents := make(map[protobuf.CommunityEvent_EventType]bool)
+	for _, event := range o.config.EventsData.Events {
+		if processedEvents[event.Type] {
```

Comment:
I think that's wrong, there may be multiple same type events for different users. The keys of the map should be the `memberID`.

Reply:
fixed